homeassistant/components/goodwe/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow to configure Goodwe inverters using their local API."""
+from __future__ import annotations
+
+import logging
+
+from goodwe import InverterError, connect
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_MODEL_FAMILY,
+    CONF_NETWORK_RETRIES,
+    CONF_NETWORK_TIMEOUT,
+    DEFAULT_NAME,
+    DEFAULT_NETWORK_RETRIES,
+    DEFAULT_NETWORK_TIMEOUT,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options for the component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Init object."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input: dict | None = None) -> FlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        settings_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SCAN_INTERVAL,
```

Comment:
We don't allow scan interval as option. Every integration that supports config entries allows the user to turn off automatic polling in the system options and then the user can automate the update interval as needed with the entity service `homeassistant.update_entity`.

Reply:
I'm sorry but I don't quite understand. The integration is used by very different kinds of users, some are using aggressive 1s scan interval and use it as real time energy management, many use 10s or 30s for monitoring and there are users forced to use even large 2min interval due to some technical restrictions. So it needs some user friendly way how to set it.
Could you please elaborate how to easily configure that ?
I couldn't find any guide for it in the docs and I assume most of the users would have no idea how to do that (incl. myself).

Reply:
Here's the service docs:
https://www.home-assistant.io/integrations/homeassistant/#service-homeassistantupdate_entity

Here's an example:
https://www.home-assistant.io/integrations/google_travel_time/#updating-sensors-on-demand-using-automation

Reply:
Thanks, for the example.
I don't intend to argue with you, or your design decisions, but I'm quit confused.
There are many of old integrations out there which use scan_interval and it perfectly fits the needs.
As mentioned in other review, e.g. my inverter provides me 146 sensors and I have them all displayed in one screen and I want them refreshed every 10s.
So now I need to create manual automation and list names of 146 entities instead of having single scan_interval ?
I can put this info into integration documentation, but existing users would not really like it ;-(

Is there some way how e.g to update all entities of a device so we can avoid that explicit list of  ?

Reply:
Since we use a coordinator only one service call for one entity would be needed. The coordinator will fetch new data and trigger all connected entities to update.

Reply:
scan_interval optional parameter was removed.